### PR TITLE
Moving from Semver to major.patch versioning

### PR DESCRIPTION
**Pull request by IgorGov**
Example of how versioning will look:
https://github.com/IgorGov/mizu/releases
image

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -242,14 +242,15 @@ jobs:
         id: condval
         with:
           cond: ${{ github.ref == 'refs/heads/main' }}
-          if_true: "minor"
-          if_false: "patch"
+          if_true: "official"
```
Comment (nimrod-up9) [inline .github/workflows/release.yml:245]: Rename to "stable"? The dev versions are also official.

**Comment by gadotroee** [comment]
I converted it to draft because two things need to be chnaged and it is not ready to for merge.
1. semver class https://github.com/up9inc/mizu/blob/develop/shared/semver/semver.go (it will not support those new versions format)
2. all calls to Auto Increment should use the new step.

File: cli/mizu/version/versionCheck.go [hunk 2]
```diff
@@ -12,31 +12,30 @@ import (
 
 	"github.com/up9inc/mizu/cli/apiserver"
 	"github.com/up9inc/mizu/cli/mizu"
+	"github.com/up9inc/mizu/cli/pkg/version"
 	"github.com/up9inc/mizu/shared/logger"
 
 	"github.com/google/go-github/v37/github"
 	"github.com/up9inc/mizu/cli/uiUtils"
-	"github.com/up9inc/mizu/shared/semver"
```
Comment (RoyUP9) [inline cli/mizu/version/versionCheck.go:19]: is this in any use anymore?

File: cli/mizu/version/versionCheck.go [hunk 3]
```diff
@@ -12,31 +12,30 @@ import (
...
-	apiSemVer, err := apiServerProvider.GetVersion()
+	apiVer, err := apiServerProvider.GetVersion()
 	if err != nil {
 		return false, err
 	}
 
-	if !semver.SemVersion(apiSemVer).IsValid() {
-		logger.Log.Errorf(uiUtils.Red, fmt.Sprintf("api version (%s) is not a valid SemVer", apiSemVer))
-		return false, nil
+	equals, err := version.AreEquals(apiVer, mizu.Ver)
```
Comment (RoyUP9) [inline cli/mizu/version/versionCheck.go:28]: can be changed to one liner with if and else if

File: cli/mizu/version/versionCheck.go (hunk 2 above)
Comment (IgorGov) [reply cli/mizu/version/versionCheck.go:19]: yes, k8s versions check

File: cli/pkg/version/version.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	major, err := strconv.Atoi(match[1])
+	if err != nil {
+		return nil, fmt.Errorf("error parsing major int: %s, err %w", match[1], err)
+	}
+	patch, err := strconv.Atoi(match[2])
+	if err != nil {
+		return nil, fmt.Errorf("error parsing patch int: %s, err %w", match[2], err)
+	}
+
+	if match[3] == "" {
```
Comment (RoyUP9) [inline cli/pkg/version/version.go:30]: maybe change to check to len(match) == 3?

File: cli/pkg/version/version_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
+package version
+
+import (
+	"testing"
+)
+
+func TestEqualsEquality(t *testing.T) {
+	first, second := "1.0", "1.0"
```
Comment (RoyUP9) [inline cli/pkg/version/version_test.go:8]: you can use test cases and pass it different version checks each time

File: cli/pkg/version/version.go (hunk 4 above)
Comment (IgorGov) [reply cli/pkg/version/version.go:30]: it will always will be 4

File: cli/pkg/version/version_test.go [hunk 6]
```diff
@@ -5,77 +5,100 @@ import (
...
-		t.Fatalf("Expected %s > %s", first, second)
+	tests := []struct {
+		Name   string
+		First  string
+		Second string
+	}{
+		{Name: "major", First: "2.0", Second: "1.0"},
+		{Name: "patch", First: "1.1", Second: "1.0"},
+		{Name: "incremental", First: "1.0-dev1", Second: "1.0-dev0"},
+		{Name: "major", First: "1.0", Second: "1.0-dev1"},
```
Comment (RoyUP9) [inline cli/pkg/version/version_test.go:73]: i think you can't have 2 test cases with the same name

File: cli/pkg/version/version_test.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			if greater, _ := GreaterThen(test.First, test.Second); !greater {
+				t.Fatalf("Expected %s > %s", test.First, test.Second)
+			}
+		})
+	}
+}
+
+func TestGreaterThenLessthen(t *testing.T) {
```
Comment (RoyUP9) [inline cli/pkg/version/version_test.go:84]: Rename to TestGreaterThenLessThen

**Review by RoyUP9 — APPROVED** [review]

**Pull request merged**